src/plugins/query_enhancements/server/routes/index.ts

```diff
@@ -12,151 +12,130 @@ import {
-          query: schema.object({
-            query: schema.string(),
-            language: schema.string(),
-            dataset: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-            format: schema.string(),
+export function defineSearchStrategyRouteProvider(logger: Logger, router: IRouter) {
+  /**
+   * @param searchStrategies - The available search strategies.
+   * @param searchStrategyId - The ID of the search strategy to use.
+   *
+   * @example
+   * API Request Body:
+   * ```json
+   * {
+   *   "query": {
+   *     "query": "SELECT * FROM my_index",
+   *     "language": "sql",
+   *     "dataset": {
+   *       "id": "my_dataset_id",
+   *       "title": "My Dataset"
+   *     },
+   *     "format": "json"
+   *   },
+   *   @experimental
+   *   "aggConfig": {
+   *     // Optional aggregation configuration
+   *   },
+   *   @deprecated
+   *   "df": {
+   *     // Optional data frame configuration
+   *   }
+   * }
+   * ```
+   */
+  return function (
+    searchStrategyId: string,
+    searchStrategy: ISearchStrategy<IOpenSearchDashboardsSearchRequest, IDataFrameResponse>
+  ) {
+    const path = `${API.SEARCH}/${searchStrategyId}`;
+    router.post(
+      {
+        path,
+        validate: {
+          body: schema.object({
+            query: schema.object({
+              query: schema.string(),
+              language: schema.string(),
+              dataset: schema.nullable(schema.object({}, { unknowns: 'allow' })),
+              format: schema.string(),
+            }),
+            aggConfig: schema.nullable(schema.object({}, { unknowns: 'allow' })),
+            df: schema.nullable(schema.object({}, { unknowns: 'allow' })),
           }),
-          aggConfig: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-          df: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-        }),
+        },
       },
-    },
-    async (context, req, res): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
-      try {
-        const queryRes: IDataFrameResponse = await searchStrategies[searchStrategyId].search(
-          context,
-          req as any,
-          {}
-        );
-        return res.ok({ body: { ...queryRes } });
-      } catch (err) {
-        return res.custom({
-          statusCode: err.name,
-          body: err.message,
-        });
+      async (context, req, res): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+        try {
+          const queryRes: IDataFrameResponse = await searchStrategy.search(context, req as any, {});
+          return res.ok({ body: { ...queryRes } });
+        } catch (err) {
+          return res.custom({
+            statusCode: err.name,
+            body: err.message,
+          });
+        }
       }
-    }
-  );
+    );
 
-  router.get(
-    {
-      path: `${path}/{queryId}`,
-      validate: {
-        params: schema.object({
-          queryId: schema.string(),
-        }),
+    router.get(
```

Comment:
headsup git might get confused as i plan on deleting most of these unused methods here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8226

Reply:
Do you want to decide which PR goes in first?

Reply:
resolved the conflicts since 8226 went in